build.gradle

```diff
@@ -98,6 +116,25 @@ test {
     }
 }
 
+task integTest(type: Test) {
+    group = LifecycleBasePlugin.VERIFICATION_GROUP
+    description = 'Runs the integration tests.'
+
+    testClassesDir = sourceSets.integTest.output.classesDir
+    classpath = sourceSets.integTest.runtimeClasspath
+
+    binResultsDir = file("$buildDir/integration-test-results/binary/integTest")
+
+    reports {
+        html.destination = "$buildDir/reports/integration-test"
+        junitXml.destination = "$buildDir/integration-test-results"
```

Comment:
It might be better to keep the `integTest` task reports parallel to the `test` task reports.  For example, the following destinations follow the convention set by the `test` task:

```gradle
task integTest(type: Test) {
    // . . .
    reports {
        html.destination = "$buildDir/reports/tests/integTest"
        junitXml.destination = "$buildDir/test-results/integTest"
    }
}
```

(I named the folders `integTest` here based on my comment above, but call them whatever you like.)

If you do choose to have the folder names match the task name, this snippet might be easier:

```gradle
task integTest(type: Test) {
    // . . .
    reports.html.destination = file("${reports.html.destination}/$name")
    reports.junitXml.destination = file("${reports.junitXml.destination}/$name")
}
```

This uses Gradle defaults as much as possible to keep things DRY.